src/Tests/VCEL.Test/FunctionTests.cs

```diff
@@ -1,384 +1,515 @@
+    [TestCase("reverse('')", "")]
+    [TestCase("replace('test_replace', '_replace', '')", "test")]
+    [TestCase("replace('test_replace', '_replace', '_test')", "test_test")]
+    [TestCase("replace(null, '_replace', '_test')", null)]
+    [TestCase("replace('test_replace', null, '_test')", "test_replace")]
+    [TestCase("replace('test_replace', '_replace', null)", "test")]
+    [TestCase("replace(null, null, null)", null)]
+    [TestCase("get('test_get', 0)", "t")]
+    [TestCase("get('test_get', 1)", "e")]
+    [TestCase("get('test_get', 6)", "e")]
+    [TestCase("get('test_get', -1)", "t")]
+    [TestCase("get('test_get', -2)", "e")]
+    [TestCase("get('test_get', 100)", null)]
+    [TestCase("get('test_get', -100)", null)]
+    [TestCase("get(null, 0)", null)]
+    [TestCase("get(null, null)", null)]
+    [TestCase("get('test_get', null)", null)]
+    [TestCase("get(null, 'test_get')", null)]
+    [TestCase("get('test_get', 'test_get')", null)]
+    public void EvalStringFunctions(string exprString, object expected)
+    {
+        foreach (var parseResult in CompositeExpression.ParseMultiple(exprString))
         {
-            foreach (var parseResult in CompositeExpression.ParseMultiple(exprString))
+            var expr = parseResult.Expression;
+            if (parseResult is { Success: false } or { ParseErrors.Count: > 0 })
             {
-                var expr = parseResult.Expression;
-                var result = expr.Evaluate(new { });
-                Assert.That(result, Is.EqualTo(expected));
+                Assert.Fail(
+                    $"Failed to parse expression: {exprString}{Environment.NewLine}{Environment.NewLine}{string.Join(Environment.NewLine, parseResult.ParseErrors.Select(e => e.Message))}");
             }
+
+            var result = expr.Evaluate(new { });
+            Assert.That(result, Is.EqualTo(expected));
         }
+    }
 
-        [TestCase("lowercase(A)", "ABCD", "abcd")]
-        [TestCase("uppercase(A)", "abcd", "ABCD")]
-        [TestCase("substring(A, 5)", "test_substring", "substring")]
-        [TestCase("substring(A, 0, 4)", "test_substring", "test")]
-        [TestCase("split(A, '_')", "test_split", new string[] { "test", "split" })]
-        [TestCase("replace(A, '_replace', '')", "test_replace", "test")]
-        [TestCase("replace(A, '_replace', '_test')", "test_replace", "test_test")]
-        public void EvalStringFunctions_Property(string exprString, string source, object expected)
+    [TestCase("length([])", 0)]
+    [TestCase("length([1, 2, 3])", 3)]
+    [TestCase("length([null])", 1)]
+    [TestCase("length(null)", null)]
+    [TestCase("length([...[1], 2)", 2)]
+    [TestCase("contains([1, 2, 3], 1)", true)]
+    [TestCase("contains([1, 2, 3], 4)", false)]
+    [TestCase("contains([1, 2, 3], null)", false)]
+    [TestCase("contains(null, 1)", false)]
+    [TestCase("contains(null, null)", false)]
+    [TestCase("startsWith([1, 2, 3], 1)", true)]
```

Comment:
feels a bit weird to use startswith on array.

this together with "split" can do a lot of things

Reply:
I'm undecided, but wonder what happens when you do `startsWith([1, 2, 3], [1, 2])`.

Reply:
Not supported right now, but I think a possible future improvement